src/odemis/dataio/tiff.py

```diff
@@ -1795,13 +1795,22 @@ def _saveAsMultiTiffLT(filename, ldata, thumbnail, compressed=True, multiple_fil
         # Only get the corresponding data for this file
         ldata = sorted_x[file_index][1]
 
+    # Extract ImageJ compatible Image description
+    num_channels = len(ldata)
+    try:
+        _, _, num_slices, height, width = ldata[0].shape
+    except:
+        num_slices = 1
```

Comment:
That is very very rough. I would suggest to at least ensure that it's true (ie, that all the channels have the same shape.
At ImageJ metadata cannot describe all the type of data we support, you have (at least) two options:
* Describe only the first images for which this is correct
* Only put ImageJ metadata if it's completely correct

I'd go for the first option.
I'd suggest that in the code, you check the ZYX dimension of the first DataArray, and you add as "Channels" the other DataArray which have the same ZYX dimension. You could go more fancy and if there is a C dimension, you report it as channels.

Reply:
@pieleric That is true indeed, I have done slight changes to the code as below    
`    try:
        num_slices = ldata[0].shape[-3]
    except:
        num_slices = 1`
I am not able to gauge, in which cases will the above code will fail to incorporate your suggestion to improve this block.